openverse_catalog/dags/providers/provider_api_scripts/smithsonian.py

```diff
@@ -202,16 +202,33 @@ def get_new_and_outdated_unit_codes(
     return new_unit_codes, outdated_unit_codes
 
 
+def get_unit_codes_from_api(
+    units_endpoint: str = UNITS_ENDPOINT, retries: int = 3
+) -> set:
+    query_params = {"api_key": API_KEY, "q": "online_media_type:Images"}
+    unit_codes_from_api = set()
+
+    if retries > 0:
+        response_json = delayed_requester.get_response_json(
```

Comment:
Does it return an otherwise correct but simply empty response?
`delayed_requester.get_response_json` already has the functionality of retrying if the response is None. It might be possible to add an optional condition to `get_response_json` so that it checks for `terms` not being empty instead of the response being non-None, and reusing that.

Reply:
> Does it return an otherwise correct but simply empty response?

That's correct Olga, it returns a healthy response of an empty array.

Reply:
Yep, unfortunately! We're getting a `200` status with `terms: []` 🙃 Since the response is completely healthy and even contains the `terms` array that we want, this problem is very specific to Smithsonian and I think `get_response_json` probably should not be aware of it. It is very strange!